### PR TITLE
feat(superuser): Require u2f on superuser access form BE

### PR DESCRIPTION
**Pull request by maxiuyuan**
As discussed in the security Updates email, we will now require u2f on the superuser access form.

We will now validate the user's u2f device through the superuser modal on sentry saas. Otherwise, we would check the user password or u2f on self-hosted or on-prem.

A feature flag is also added for quick rollback, if rolled back, sentry saas will not check password or u2f but only sso session.

accidentally merged https://github.com/getsentry/sentry/pull/38534 into https://github.com/getsentry/sentry/pull/37953, so I needed to the feature into the fe and be prs again

### CODE REVIEW

File: src/sentry/api/endpoints/auth_index.py [hunk 1]
```diff
@@ -97,34 +108,13 @@ def _validate_superuser(self, validator, request):
...
-                user_id=request.user.id, type=U2fInterface.type
-            ).exists():
-                return True
-            return False
-
-        if _require_password_or_u2f_check():
+        if not DISABLE_SSO_CHECK_SU_FORM_FOR_LOCAL_DEV and verify_authenticator or is_self_hosted():
             authenticated = self._verify_user_via_inputs(validator, request)
+        else:
+            authenticated = True
```
Comment (leeandher) [inline src/sentry/api/endpoints/auth_index.py:117]: nit: Can replace this this with ternary assignment if you want

File: src/sentry/features/__init__.py [hunk 2]
```diff
@@ -147,6 +147,8 @@
 default_manager.add("organizations:widget-library", OrganizationFeature, True)
 default_manager.add("organizations:widget-viewer-modal", OrganizationFeature, True)
 default_manager.add("organizations:widget-viewer-modal-minimap", OrganizationFeature, True)
+default_manager.add("organizations:dashboard-custom-measurement-widgets", OrganizationFeature, True)
```
Comment (leeandher) [inline src/sentry/features/__init__.py:150]: Is this feature flag (Line 150) meant to be here?

**Review by leeandher — COMMENTED** [review]
Looks good, thought I think a rogue feature flag snuck its way into this PR maybe? (`__init__.py`)

File: src/sentry/features/__init__.py (hunk 2 above)
Comment (maxiuyuan) [reply src/sentry/features/__init__.py:150]: hm i dont think so, ill get rid of it ty!

**Review by leeandher — APPROVED** [review]

**Pull request merged**